web/src/db.tsx

```diff
@@ -212,23 +254,42 @@ export interface WgsCovPlotDB {
   /** list of virus segment names if segmented virus being shown */
   segments?: string[];
 
+  /** Max length among samples of 1 segment**/
+  maxSegmentLength?: MaxSegmentLength
+
   /** Show amplicon features? */
-  show_amplicons: boolean;
+  show_amplicons?: boolean;
+  /** Show primer matches? */
+  show_primer_matches?: boolean;
   /** Show gene features? */
   show_genes: boolean;
   /** ECharts object for variant calling heatmap */
   variantHeatmap?: ECharts | EChartsType;
   /** variant calls */
-  variants?: SampleVariantCalls | SampleSegmentVariantCalls;
+  variants: SampleVariantCalls | SampleSegmentVariantCalls;
```

Comment:
@nhhaidee Is variant calling data always expected to be present? What if there's only a BAM file in the output directory?